com.ibm.streamsx.topology/opt/python/include/splpy_funcop.h

```diff
@@ -114,6 +117,58 @@ class SplpyFuncOp : public SplpyOp {
                "streamsx.topology.runtime", wrapfn, appCallable, extraArg));
       }
 
+      /**
+       * Register a state handler for the operator.  The state handler
+       * handles checkpointing and supports consistent regions.  Checkpointing
+       * will be enabled only if the operator is stateful and if it can be
+       * saved and restored using dill.
+       */
+      void setupStateHandler() {
+        // If the value of the pyStateful param is true, create and register
+        // a state handler instance.  Otherwise, register a do-nothing
+        // state handler.
+        SPL::boolean stateful = static_cast<SPL::boolean>(op()->getParameterValues("pyStateful")[0]->getValue());
+        assert(!stateHandler);
+        // Ensure that callable() can be pickled before using it in a state
+        // handler.
+        PyObject * pickledCallable = NULL;
+        if (stateful) {
+          SplpyGIL lock;
+          PyObject * dumps = SplpyGeneral::loadFunction("dill", "dumps");
+          PyObject * args = PyTuple_New(1);
+          Py_INCREF(callable());
+          PyTuple_SET_ITEM(args, 0, callable());
+          pickledCallable = PyObject_CallObject(dumps, args);
+          Py_DECREF(args);
+          Py_DECREF(dumps);
+
+          if (!pickledCallable) {
+            // the callable cannot be pickled
+            // continue without checkpointing for this operator
+            if (PyErr_Occurred()) {
+              PyObject * type = NULL;
+              PyObject * value = NULL;
+              PyObject * traceback = NULL;
+
+              PyErr_Fetch(&type, &value, &traceback);
+              if (value) {
+                SPL::rstring text;
+                // note pyRStringFromPyObject returns zero on success
+                if (!pyRStringFromPyObject(text, value)) {
+                  SPLAPPTRC(L_WARN, "Checkpointing is disabled for the " << op()->getContext().getName() << " operator because of python error " << text, "python");
```

Comment:
Seems like this is an error, not just a warning.

Reply:
Really?  If someone enables checkpointing in a topology with an operator that cannot be saved, we don't allow it to run at all?

Reply:
Well, it's not fulfilling the required functionality. Seems safer to fail rather than continue on in silence not checkpointing.

Reply:
I updated to log an error and throw an exception, shutting down the operator.